cranelift-simplejit/src/backend.rs

```diff
@@ -150,6 +150,11 @@ pub struct SimpleJITCompiledData {
     relocs: Vec<RelocRecord>,
 }
 
+/// A handle to allow freeing memory allocated by the `Backend`.
+pub struct SimpleJITMemoryHandle {
+    simplejit: SimpleJITBackend,
```

Comment:
This will keep some unnecessary fields alive: https://github.com/Reisz/cranelift/blob/51b9e7e781029f966a57567852e4feafb3905692/cranelift-simplejit/src/backend.rs#L118-L120

Please put the `*_memory` fields directly in here instead.

Reply:
What do you think about the following:
```rust
pub struct SimpleJITBackend {
    isa: Box<dyn TargetIsa>,
    symbols: HashMap<String, *const u8>,
    libcall_names: Box<dyn Fn(ir::LibCall) -> String>,
    memory: SimpleJITMemoryHandler,
}

pub struct SimpleJITMemoryHandle {
    code: Memory,
    readonly: Memory,
    writable: Memory,
}
```
 That would keep the code simpler.

Reply:
LGTM